ccm_web/server/src/api/api.section.handler.ts

```diff
@@ -47,10 +52,19 @@ export class SectionApiHandler {
     return enrollmentsResult.map(e => e.user.login_id)
   }
 
-  async enrollUser (user: SectionUserDto): Promise<CanvasEnrollment | APIErrorData> {
-    const enrollLoginId = user.loginId
-      .replace(/@([^@.]+\.)*umich\.edu$/gi, '')
-      .replace('@', '+')
+  async enrollUser (user: SectionUserDto | SectionExternalUserDto): Promise<CanvasEnrollment | APIErrorData> {
+    let userId: string
+    let enrollId: string
+    if (user instanceof SectionUserDto) {
+      userId = user.loginId
+      enrollId = userId
+        .replace(/@([^@.]+\.)*umich\.edu$/gi, '')
+        .replace('@', '+')
+    } else {
+      userId = user.email
+      enrollId = userId
+        .replace('@', '+')
+    }
```

Comment:
Would there be any benefit to moving this branching logic out of this method? I think I would probably not rely on the DTO object here and accept two parameters, `loginId` and `enrollmentType`. Then the branching logic could happen higher up, maybe at the `api.service.ts` level.

Reply:
To be clear, this can be considered optional, I just found it be an intriguing code structure problem.

Reply:
Sure.  I can see that as a good way to simplify the code.

What would you recommend for the replacement of umich.edu strings in the user ID when appropriate?  That currently depends on the kind of user object being passed in.  Would you pass an additional argument indicating the replacement should be performed?  Would you move that replacement operation from `enrollUser()` up to `enrollUsers()` (note: plural) or higher?

Reply:
Could the replacement operation happen at the `api.service.ts` level, as suggested before? There I believe we'll differentiate between external and internal enrollments and then call the `SectionApiHandler`, so you could just apply whatever logic you wanted to each type of user's `loginId`/`email` and then proceed to use the same `SectionApiHandler` method.

Reply:
This is obviated by the change in architecture (separate endpoints for creating the user and enrolling them). Resolving...